networkx/algorithms/link_prediction.py

```diff
@@ -2,9 +2,11 @@
 Link prediction algorithms.
 """
 
-
 from math import log
 
+import scipy as sp
+from scipy import sparse
+
```

Comment:
You should put the scipy import statements inside the functions that need it, so the module can still be loaded even if scipy is not installed. 

For the tests, you should use the `pytest.importorskip` function (pronounced import-or-skip) so that the test module gets skipped (if all functions tested require numpy), or so individual tests get skipped when the needed package is not installed.  Something like:
```python
np = pytest.importorskip("numpy")
```

Reply:
Thanks, just made those changes